osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/SecurityGroupMemberDeleteTask.java

```diff
@@ -160,4 +141,66 @@ public String getName() {
         return LockObjectReference.getObjectReferences(this.sgm.getSecurityGroup());
     }
 
+    private void deleteFromDB(EntityManager em , VMPort port) {
+        this.log.info("Removing port " + port);
+        VM vm = port.getVm();
+        if (vm != null) {
+            vm.removePort(port);
+            port.setVm(null);
+        }
+
+        Network network = port.getNetwork();
+        if (network != null) {
+            network.removePort(port);
+        }
+
+        for (DistributedApplianceInstance dai : port.getDais()) {
+            dai.removeProtectedPort(port);
+            port.removeDai(dai);
+        }
+
+        OSCEntityManager.delete(em, port, this.txBroadcastUtil);
+    }
+
+    private void deleteFromDB(EntityManager em, VM vm) {
+        this.log.info("Removing vm " + vm);
+
+        for (VMPort port : vm.getPorts()) {
+            deleteFromDB(em, port);
+        }
+
+        OSCEntityManager.delete(em, vm, this.txBroadcastUtil);
+    }
+
+    private void deleteFromDB(EntityManager em, Network network) {
+        Set<VMPort> ports = new HashSet<>(network.getPorts());
+        for (VMPort vmPort : ports) {
+            // If the VM was created on behalf of this port, delete it.
+            VM vm = vmPort.getVm();
+            deleteFromDB(em, vmPort);
+            if (vm != null) {
+                vm.removePort(vmPort);
```

Comment:
isnt this done twice once in the "deleteFromDB(em, vmPort)" call? i think this would be a noop.